tls-tutorial/doc/use_your_own_certificate.md

```diff
@@ -0,0 +1,685 @@
+
+The most important part here are the cipher and the ticket lifetime of 5 minutes.
+
+**Side-note:** if you ever experience TLS connections breaking of after a certain time, you should check first this session ticket lifetime and then any intermediate network components like firewalls. Sometimes they interfere with the connection renewal process.
+
+Finally, you can connect the MySQL client to the server using TLS:
+
+```shell
+mysql -u tutorial_user -p --ssl-mode=REQUIRED -u tutorial_user
+```
+
+The source side of our import via TLS is now set up and ready to use.
+
+## Setting up Docker
+
+For the next steps we need a Docker setup
+
+1. Install Docker
+    ```shell
+    sudo apt install docker.io
+    ```
+2. Add your current user to the `docker` group
+    ```shell
+    sudo usermod -aG docker "$USER"
+    ```
+3. Log out and back in to apply the group change
+4. Verify that your current user is in the `docker` group
+    ```shell
+   groups
+    ```
+
+## Installing Exasol via Docker
+
+Exasol's [docker-db](https://github.com/exasol/docker-db) provides a Docker container that is quite convenient for experiments like our tutorial.
+
+The following docker command downloads the container's layers and afterwards starts the container. Expect a couple of minutes until the download and database start are done.
+
+```shell
+docker run --name exasoldb -p 8563:8563 -p 2580:2580 --detach --privileged --stop-timeout 120  exasol/docker-db:7.1.17
+```
+
+There are two ports forwarded to the host:
+
+- 8563: database port
+- 2580: [BucketFS] port
+
+### Installing the MySQL JDBC driver in Exasol
+
+[Import](https://docs.exasol.com/db/latest/sql/import.htm) into Exasol requires that the database driver of the source database (MySQl in our case) is installed in Exasol.
+
+Installation in the Docker variant requires uploading the driver to a [BucketFS] bucket and updating a configuration file.
+
+1. Download the [MySQL Connector] (JDBC driver)
+   ```shell
+   wget https://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-j-8.0.31.tar.gz
+   ```
+2. Get the ID of the Exasol Docker container
+   ```shell
+   container_id=$(docker ps -a | grep exasol | sed -e's/ .*//')
+   ```
+3. Get the write-password of the default Bucket in BucketFS
+   ```shell
+   write_pwd=$(docker exec -it "$container_id" cat /exa/etc/EXAConf | grep WritePasswd | sed -e's/^.* = //' -e's/[\n\r]//g' | base64 --decode)
+   ```
+4. Copy the driver to the default Bucket
+   ```shell
+   curl -vX PUT -T mysql-connector-j-8.0.31/mysql-connector-j-8.0.31.jar "http://w:$write_pwd@localhost:2580/default/drivers/jdbc/mysql-connector-j-8.0.31.jar"
+   ```
+5. Create a driver configuration (`settings.cfg`)
+   ```shell
+   echo 'DRIVERNAME=MYSQL_JDBC
+   JAR=mysql-connector-j-8.0.31.jar
+   DRIVERMAIN=com.mysql.cj.jdbc.Driver
+   PREFIX=jdbc:mysql:
+   NOSECURITY=YES
+   FETCHSIZE=100000
+   INSERTSIZE=-1' > settings.cfg
+   ```
```

Comment:
Make sure that file has no leading spaces.

Reply:
That is a pretty good point. While in the rendered version there are no leading spaces, people get them when copying from the source.

Reply:
Added disclaimer.